ssc/cmod_utilityrate5.cpp

```diff
@@ -1798,8 +1796,15 @@ class cm_utilityrate5 : public compute_module
 						if (excess_monthly_dollars || rollover_credit)
 						{
 //							monthly_cumulative_excess_dollars[m] = -monthly_bill[m];
-							dollars_applied += monthly_cumulative_excess_dollars[m] + monthly_ec_charges[m];
-							monthly_cumulative_excess_dollars[m] = -monthly_ec_charges[m];
+                            if (m != net_metering_credit_month)
```

Comment:
Is this balancing out in the utility bill reported by SAM? Do you have a reference SAM project file for testing? Do you check for negative cumulative excess dollars for each month? I think we need both an ssc test and a SAM project file for validation using rates with dollar rollovers, no excess dollars, and some months with and some without excess dollars to fully validate.

Reply:
Yes, I'm kicking myself for not doing more validation on this before the release.

Here's the SAM file for the SSC tests that were affected by this PR: 
[net_billing_monthly_variation.zip](https://github.com/NREL/ssc/files/5895343/net_billing_monthly_variation.zip)

This is the file that Paul sent me that caught the net metering with $ credits bug initially - we expect net metering credits to align with any negative bills, which they do after this PR: 
[net-metering-$-credits-test-sam-2020-11-29.zip](https://github.com/NREL/ssc/files/5895358/net-metering-.-credits-test-sam-2020-11-29.zip)

And for net billing: 
[monthly-billing-results-tests-2021-11-29.zip](https://github.com/NREL/ssc/files/5895367/monthly-billing-results-tests-2021-11-29.zip)

I'll post Excel files when I have them. Once I've used Excel to validate the existing tests (and find any other bugs), we can determine whether the existing load profiles hit all of the relevant cases or whether we need additional load profiles to fully validate.

Reply:
This file: 
[net_metering_monthly_variation_match_excel.zip](https://github.com/NREL/ssc/files/5895625/net_metering_monthly_variation_match_excel.zip)

matches https://github.com/NREL/SAM-documentation/blob/master/Unit%20Testing/Utility%20Rates/SAM%202016.3.14%20Test%20Files%20with%20Spreadsheets/Tiered%20TOU%20single%20meter%20rollover%20%24.xlsx, with that caveat that "Net Metering Credits" now corresponds to "net metering credits applied in month m"

Reply:
Ok, it looks like I introduced the bug here: https://github.com/NREL/ssc/commit/f94e50429a1bcb330a7a33af6679f930c7fbb2cf#diff-b252d497c315d490c9e716fbc1355f21e42f51db78299f43e7935b068bb03716

I wrote an Excel file for the December case of net_billing_monthly_variation.zip here: 
[2020.11.29_net_billing_carryover_dec.xlsx](https://github.com/NREL/ssc/files/5896471/2020.11.29_net_billing_carryover_dec.xlsx)

I'll get the December case fixed up, then see how that affects the other cases - it might be additional columns on the above Excel can represent the true-up month switches.

For what it's worth, that number matches the number in "Electricity bill with system ($/yr)" in 2020.2.29, but not "Electricity bill with system ($)" - so there might be another bug lurking somewhere.

Reply:
@brtietz, thanks for adding all of these validation tests and supporting documentation. This has been sorely missing from way back before you got here. If anyone is to be kicking themself, it is me!